crates/forge-runner/src/lib.rs

```diff
@@ -277,19 +285,71 @@ fn choose_test_strategy_and_run(
     runner: Arc<SierraCasmRunner>,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
-    send: &Sender<()>,
-) -> JoinHandle<Result<TestCaseSummary>> {
+    send: Sender<()>,
+) -> JoinHandle<Result<AnyTestCaseSummary>> {
     if args.is_empty() {
-        run_test(case, runner, runner_config, runner_params, send.clone())
+        tokio::task::spawn(async move { 
+            let res = run_test(case, runner, runner_config, runner_params, send).await??;
+            Ok(AnyTestCaseSummary::Single(res)) 
+        })
     } else {
-        run_with_fuzzing(
-            args,
-            case,
-            runner,
-            runner_config,
-            runner_params,
-            send.clone(),
-        )
+        tokio::task::spawn(async move { 
+            let res = run_with_fuzzing(
+                args,
+                case,
+                runner,
+                runner_config,
+                runner_params,
+                send,
+            ).await??;
+            Ok(AnyTestCaseSummary::Fuzzing(res)) 
+        })
+
+    }
+}
+
+fn fuzzing_run_summary(results: Vec<TestCaseSummary<Single>>) -> Result<TestCaseSummary<Fuzzing>>  {
+    let last: TestCaseSummary<Single> = results.iter().last().cloned().expect("Fuzz test should always run at least once");
+    match last {
+        TestCaseSummary::Passed { name, msg, arguments, gas_info: _, test_statistics: _, latest_block_number} => {
+            let runs = results.len();
+            let gas_usages_vec = results
+                .into_iter()
+                .filter(|item| matches!(item, TestCaseSummary::Passed { .. }))
+                .map(|a| match a {
+                    TestCaseSummary::Passed { gas_info, .. } => gas_info,
+                    _ => unreachable!(),
+                });
+
+            let max = gas_usages_vec
+                .clone()
```

Comment:
@tkumor3 you were right here we probably cannot do much to prevent cloning here.